src/test/java/com/google/sps/AlgorithmTest.java

```diff
@@ -0,0 +1,293 @@
+    placeToVisitA.setProperty("category", "places-to-visit");
+    placeToVisitA.setProperty("description", "A great place to visit.");
+    placeToVisitA.setProperty("cost-rating", 3);
+    placeToVisitA.setProperty("crowd-rating", 3);
+    Entity placeToVisitB = new Entity("Recommendation");
+    placeToVisitB.setProperty("place-name", "Place To Visit B");
+    placeToVisitB.setProperty("latitude", 40.3);
+    placeToVisitB.setProperty("longitude", -0.3);
+    placeToVisitB.setProperty("category", "places-to-visit");
+    placeToVisitB.setProperty("description", "A lovely place to visit.");
+    placeToVisitB.setProperty("cost-rating", 5);
+    placeToVisitB.setProperty("crowd-rating", 4);
+    Entity placeToVisitC = new Entity("Recommendation");
+    placeToVisitC.setProperty("place-name", "Place To Visit C");
+    placeToVisitC.setProperty("latitude", 40.4);
+    placeToVisitC.setProperty("longitude", -0.4);
+    placeToVisitC.setProperty("category", "places-to-visit");
+    placeToVisitC.setProperty("description", "A nice place to visit.");
+    placeToVisitC.setProperty("cost-rating", 4);
+    placeToVisitC.setProperty("crowd-rating", 5);
+    Entity placeToVisitD = new Entity("Recommendation");
+    placeToVisitD.setProperty("place-name", "Place To Visit D");
+    placeToVisitD.setProperty("latitude", 40.5);
+    placeToVisitD.setProperty("longitude", -0.5);
+    placeToVisitD.setProperty("category", "places-to-visit");
+    placeToVisitD.setProperty("description", "An interesting place to visit.");
+    placeToVisitD.setProperty("cost-rating", 2);
+    placeToVisitD.setProperty("crowd-rating", 3);
+    Entity placeToVisitE = new Entity("Recommendation");
+    placeToVisitE.setProperty("place-name", "Place To Visit E");
+    placeToVisitE.setProperty("latitude", 40.6);
+    placeToVisitE.setProperty("longitude", -0.6);
+    placeToVisitE.setProperty("category", "places-to-visit");
+    placeToVisitE.setProperty("description", "A charming place to visit.");
+    placeToVisitE.setProperty("cost-rating", 1);
+    placeToVisitE.setProperty("crowd-rating", 2);
+    Entity placeToVisitF = new Entity("Recommendation");
+    placeToVisitF.setProperty("place-name", "Place To Visit F");
+    placeToVisitF.setProperty("latitude", 40.7);
+    placeToVisitF.setProperty("longitude", -0.7);
+    placeToVisitF.setProperty("category", "places-to-visit");
+    placeToVisitF.setProperty("description", "A wonderful place to visit.");
+    placeToVisitF.setProperty("cost-rating", 2);
+    placeToVisitF.setProperty("crowd-rating", 1);
+    // Store recommendations in datastore
+    ds.put(restaurantA);
+    ds.put(restaurantB);
+    ds.put(restaurantC);
+    ds.put(restaurantD);
+    ds.put(restaurantE);
+    ds.put(studyPlaceA);
+    ds.put(studyPlaceB);
+    ds.put(placeToVisitA);
+    ds.put(placeToVisitB);
+    ds.put(placeToVisitC);
+    ds.put(placeToVisitD);
+    ds.put(placeToVisitE);
+    ds.put(placeToVisitF);
```

Comment:
Nit: This whole part looks busy and repetitive. I am not sure but maybe having a static method like the following would help?

```
static void storeRecommendation(String placeName, double latitude, ... , double crowdRating) {
  Entity entity = new Entity("Recommendation");
  entity.setProperty("place-name", placeName);
  .
  .
  .
  ds.put(entity);
}
``` 

Totally up to you. If having this method doesn't make this part look better and more readable, then there is no need to have it.

Reply:
Good idea - I tried doing this but came across lots of issues. I've added some spacing to make it easier to read, as you suggested, instead.